staging/src/k8s.io/client-go/tools/cache/reflector.go

```diff
@@ -495,3 +516,42 @@ func isExpiredError(err error) bool {
 	// check when we fully drop support for Kubernetes 1.17 servers from reflectors.
 	return apierrors.IsResourceExpired(err) || apierrors.IsGone(err)
 }
+
+// backoffManager manages jittered exponential back off with a cap and a reset period. Exponential back off will stop
+// to increase at the given max, and timer will reset after not backing off for a while
+type backoffManager struct {
```

Comment:
This seems useful enough that it should be moved to a public place imho.
client-go/util/flowcontrol seems to be a good place

Probably also a separate PR as this will be universally useful.

Reply:
thanks for the feedback! I will create a separate PR for backoff manager in `client-go/util/flowcontrol` package, and get that merged first. There will be a second PR updating reflector